go/logic/topology_recovery.go

```diff
@@ -1411,6 +1455,46 @@ func checkAndRecoverGenericProblem(analysisEntry inst.ReplicationAnalysis, candi
 	return false, nil, nil
 }
 
+// checkAndRecoverDeadGroupMemberWithReplicas checks whether action needs to be taken for an analysis involving a dead
+// replication group member, and takes the action if applicable. Notice that under our view of the world, a primary
+// replication group member is akin to a master in traditional async/semisync replication; whereas secondary group
+// members are akin to intermediate masters. Considering also that a failed group member can always be considered as a
+// secondary (even if it was primary, the group should have detected its failure and elected a new primary), then
+// failure of a group member with replicas is akin to failure of an intermediate master.
+func checkAndRecoverDeadGroupMemberWithReplicas(analysisEntry inst.ReplicationAnalysis, candidateInstanceKey *inst.InstanceKey, forceInstanceRecovery bool, skipProcesses bool) (bool, *TopologyRecovery, error) {
+	// Don't proceed with recovery unless it was forced or automatic intermediate source recovery is enabled.
+	// We consider failed group members akin to failed intermediate masters, so we re-use the configuration for
```

Comment:
> so we re-use the configuration 

but in `analysis_dao.go` it seems like you've changed that: intermediate master recovery only takes place under
```golang
if !a.IsReplicationGroupMember {
```

Reply:
What  I mean here is that we re-use  `analysisEntry.ClusterDetails.HasAutomatedIntermediateMasterRecovery` configuration to decide whether we want to fail-over group members as opposed to having a separate configuration. As mentioned in the method's doc comment, we are operating under the assumption that group secondaries with replicas are akin to intermediate masters in the sense that they perform a very similar function in the replication chain; get and apply changes from the primary (except, via GR instead of binlog), and distribute them to replicas (via the binlog). I hope this clarifies my intent.